### PR TITLE
Polish code and add side effect ut

### PR DESCRIPTION
**Pull request by 0x45f**
Polish code and add side effect ut

### CODE REVIEW

**Review by feifei-111 — DISMISSED** [review]
LGTM

**Review by Aurelius84 — DISMISSED** [review]

File: sot/opcode_translator/executor/variables/basic.py [hunk 1]
```diff
@@ -366,11 +366,33 @@ def get_value(self) -> Any:
...
+                4,
+                lambda: print(
+                    f"[Guard]: guard_fn for {self}, tracker={self.tracker.__class__.__name__}, value={frame_value_tracer.expr}"
+                ),
+            )
+            return StringifyExpression(
+                f"{frame_value_tracer.expr} == {self.get_value()}",
+                union_free_vars(frame_value_tracer.free_vars),
+            ) & StringifyExpression(
+                f"{frame_value_tracer.expr}.dtype == {self.get_value().dtype}",
```
Comment (SigureMo) [inline sot/opcode_translator/executor/variables/basic.py]:
```python
def format_dtype(dtype: np.dtype):
    return f"np.{str({dtype})}"

def format_number(number: np.number):
    return f"{format_dtype(number.dtype)}({str(number.item())})"

StringifyExpression(
    f"{frame_value_tracer.expr} == {format_number(self.get_value())}",
    union_free_vars(frame_value_tracer.free_vars, {"np": numpy}),
) & StringifyExpression(
    f"{frame_value_tracer.expr}.dtype == {format_dtype(self.get_value().dtype)}",
    union_free_vars(frame_value_tracer.free_vars, {"np": numpy})
```

**Review by feifei-111 — DISMISSED** [review]
LGTM

**Review by SigureMo — APPROVED** [review]
LGTM～

**Review by Aurelius84 — APPROVED** [review]
LGTM

**Pull request merged**